src/main/java/de/bonndan/nivio/model/Group.java

```diff
@@ -93,8 +92,13 @@ public void setColor(String color) {
         return links;
     }
 
+    /**
+     * Returns an immutable copy of the items.
+     *
+     * @return immutable copy
+     */
     public Set<Item> getItems() {
-        return items;
+        return Set.copyOf(items);
```

Comment:
You could also wrap it with `Collections.unmodifiableSet()` as modification would not make sense anymore.

Reply:
Set.of is the same, but unmodifiableSet is better to understand, so I'll change it.